### PR TITLE
fox_co decomp, Object_2F4 rename to Actor with work buffers

### PR DESCRIPTION
**Pull request by inspectredc**

### CODE REVIEW

**Review by sonicdcer — APPROVED** [review]
This is a tremendous amount of work, thank you very much for this valuable contribution!

File: src/overlays/ovl_DF4260/fox_co.c [hunk 1]
```diff
@@ -0,0 +1,3333 @@
...
+                    Item_Initialize(&gItems[i]);
+                    gItems[i].obj.status = 1;
+                    gItems[i].obj.id = OBJ_ITEM_1UP;
+                    gItems[i].obj.pos.x = gPlayer[0].pos.x + sp3C.x;
+                    gItems[i].obj.pos.y = gPlayer[0].pos.y + 100.0f;
+                    gItems[i].obj.pos.z = gPlayer[0].unk_138 + sp3C.z;
+                    gItems[i].timer_4A = 8;
+                    Object_SetInfo(&gItems[i].info, gItems[i].obj.id);
+                    func_8007B344(gItems[i].obj.pos.x, gItems[i].obj.pos.y, gItems[i].obj.pos.z, 5.0f, 0);
+                    return;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:530]:
```suggestion
                    break;
```

File: src/overlays/ovl_DF4260/fox_co.c [hunk 2]
```diff
@@ -0,0 +1,3333 @@
...
+            Math_SmoothStepToF(&boss->obj.rot.z, sp5C, 0.2f, 100.0f, 0.001f);
+        }
+        if (boss->obj.pos.y <= 10.0f) {
+            boss->obj.pos.y = 10.0f;
+        }
+        sp214 = D_DF4260_8019B6D8[17] - boss->obj.pos.x;
+        sp210 = D_DF4260_8019B6D8[18] - (boss->obj.pos.y + 300.0f);
+        sp20C = D_DF4260_8019B6D8[19] - boss->obj.pos.z;
+        sp1FC = Math_RadToDeg(Math_Atan2F(sp214, sp20C));
+        sp200 = Math_RadToDeg(-Math_Atan2F(sp210, sqrtf((sp214 * sp214) + (sp20C * sp20C))));
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:670]:
```suggestion
        sp200 = Math_RadToDeg(-Math_Atan2F(sp210, sqrtf(SQ(sp214) + SQ(sp20C))));
```

File: src/overlays/ovl_DF4260/fox_co.c [hunk 3]
```diff
@@ -0,0 +1,3333 @@
...
+            break;
+        case 16:
+            arg3->x += D_DF4260_8019B6D8[16];
+            break;
+    }
+    if ((boss->swork[arg0] & 1) || (boss->timer_05C & 1)) {
+        RCP_SetupDL_64();
+        GPC(64, 64, 255, 255);
+    }
+    return 0;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1036]:
```suggestion
    return false;
```

File: src/overlays/ovl_DF4260/fox_co.c [hunk 4]
```diff
@@ -0,0 +1,3333 @@
...
+        case 1:
+            Matrix_MultVec3f(gCalcMatrix, &sp68, &sp74);
+            D_DF4260_8019B6D8[3] = sp74.x;
+            D_DF4260_8019B6D8[4] = sp74.y;
+            D_DF4260_8019B6D8[5] = sp74.z;
+            Matrix_MultVec3f(gCalcMatrix, &sp5C, &sp74);
+            D_DF4260_8019B6D8[0] = sp74.x;
+            D_DF4260_8019B6D8[1] = sp74.y;
+            D_DF4260_8019B6D8[2] = sp74.z;
+            return;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1087]:
```suggestion
            break;
```
break should be used over return when possible.

File: src/overlays/ovl_DF4260/fox_co.c [hunk 5]
```diff
@@ -0,0 +1,3333 @@
...
+s32 func_DF4260_8018AB08(s32 arg0, Gfx** arg1, Vec3f* arg2, Vec3f* arg3, void* arg4) {
+    Actor* actor = (Actor*) arg4;
+
+    if (arg0 == 1) {
+        arg3->x += actor->fwork[1];
+    }
+    if ((arg0 == 3) && (actor->obj.id == OBJ_ACTOR_176)) {
+        arg3->x += actor->fwork[1];
+    }
+    return 0;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1143]:
```suggestion
    return false;
```

File: src/overlays/ovl_DF4260/fox_co.c [hunk 6]
```diff
@@ -0,0 +1,3333 @@
...
+            actor->unk_0CE -= actor->unk_0D6;
+            if (actor->unk_0CE <= 0) {
+                actor->obj.id = OBJ_ACTOR_179;
+                Object_SetInfo(&actor->info, actor->obj.id);
+                actor->unk_0B8 = 100;
+                actor->timer_0BC = 10;
+                actor->timer_0CA[0] = 0;
+                actor->vel.z = 0.0f;
+                actor->vel.y = 0.0f;
+                actor->vel.x = 0.0f;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1189]:
```suggestion
                actor->vel.x = actor->vel.y = actor->vel.z = 0.0f;
```
does this work?

File: src/overlays/ovl_DF4260/fox_co.c [hunk 7]
```diff
@@ -0,0 +1,3333 @@
...
+            actor->unk_0B6 = 0;
+            actor->fwork[0] += 1.0f;
+            if (actor->fwork[0] > 10.0f) {
+                actor->fwork[0] = 10.0f;
+            }
+
+            actor->iwork[0] = func_DF4260_8018AB64(actor);
+            if (actor->iwork[0] != 0) {
+                actor->unk_0B8++;
+                return;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1230]:
```suggestion
```
this shouldn't be needed.

File: src/overlays/ovl_DF4260/fox_co.c [hunk 8]
```diff
@@ -0,0 +1,3333 @@
...
+            actor->fwork[1] += 5.0f;
+            Texture_Scroll(&D_60329C0, 16, 16, 1);
+            actor->unk_0B6++;
+            if (actor->unk_0B6 >= Animation_GetFrameCount(D_602A520)) {
+                actor->unk_0B6 = 0;
+            }
+            break;
+    }
+    if (actor->iwork[1] == 0) {
+        temp_v0_2 = &gObjects80[actor->iwork[0]];
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1367]: is this temp needed?

File: src/overlays/ovl_DF4260/fox_co.c [hunk 9]
```diff
@@ -0,0 +1,3333 @@
...
+        if (((boss->unk_04E == 1) || (boss->unk_04E == 2)) && !(gFrameCount & 7)) {
+            Matrix_MultVec3f(gCalcMatrix, &D_DF4260_80199914[0], &sp84[6]);
+            Matrix_MultVec3f(gCalcMatrix, &D_DF4260_80199914[1], &sp84[7]);
+            Matrix_MultVec3f(gCalcMatrix, &D_DF4260_8019992C, &sp84[8]);
+            func_8007BC7C(sp84[6].x + boss->obj.pos.x, sp84[6].y + boss->obj.pos.y, sp84[6].z + boss->obj.pos.z, 20.0f);
+            func_8007BC7C(sp84[7].x + boss->obj.pos.x, sp84[7].y + boss->obj.pos.y, sp84[7].z + boss->obj.pos.z, 10.0f);
+            func_8007BC7C(sp84[8].x + boss->obj.pos.x, sp84[8].y + boss->obj.pos.y, sp84[8].z + boss->obj.pos.z, 10.0f);
+        }
+        if (boss->timer_052 != 0) {
+            Math_SmoothStepToF(D_801779A8, 50.0f, 1.0f, 1.6f, 0.0001f);
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1755]:
```suggestion
            Math_SmoothStepToF(&D_801779A8[0], 50.0f, 1.0f, 1.6f, 0.0001f);
```

File: src/overlays/ovl_DF4260/fox_co.c [hunk 10]
```diff
@@ -0,0 +1,3333 @@
...
+                    sp78.y = 0.0f;
+                    sp78.z = 40.0f;
+                    Matrix_MultVec3f(gCalcMatrix, &sp78, &sp6C);
+                    boss->vel.x = sp6C.x;
+                    boss->vel.y = sp6C.y;
+                    boss->vel.z = sp6C.z - D_80177D08;
+                }
+                break;
+            case 8:
+                *D_801779A8 = 20.0f;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1975]:
```suggestion
                D_801779A8[0] = 20.0f;
```

File: src/overlays/ovl_DF4260/fox_co.c [hunk 11]
```diff
@@ -0,0 +1,3333 @@
...
+                    func_800BA808(gMsg_ID_2298, RCID_PEPPY);
+                    boss->timer_052 = 350;
+                }
+                if (boss->timer_050 == 0) {
+                    boss->obj.pos.y = 900.0f;
+                    boss->unk_04E = 1;
+                    boss->fwork[13] = 180.0f;
+                    boss->fwork[12] = 1.5f;
+                    boss->fwork[6] = 800.0f;
+                    boss->fwork[3] = (gPlayer[0].camEye.z - D_80177D20) - 2500.0f;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:1776]:
```suggestion
                    boss->fwork[3] = gPlayer[0].camEye.z - D_80177D20 - 2500.0f;
```

File: src/overlays/ovl_DF4260/fox_co.c [hunk 12]
```diff
@@ -0,0 +1,3333 @@
...
+}
+
+s32 func_DF4260_8018EC54(s32 arg0, Gfx** arg1, Vec3f* arg2, Vec3f* arg3, void* arg4) {
+    if ((arg0 == 10) && (gBosses[2].unk_04E != 0) && (gBosses[3].unk_04E != 0)) {
+        *arg1 = NULL;
+    }
+    if ((arg0 != 12) && (arg0 != 10)) {
+        *arg1 = NULL;
+    }
+    return 0;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:2383]:
```suggestion
    return false;
```

File: src/overlays/ovl_DF4260/fox_co.c [hunk 13]
```diff
@@ -0,0 +1,3333 @@
...
+
+void func_DF4260_8018EF90(Boss* boss) {
+
+    Matrix_Translate(gGfxMatrix, -D_DF4260_80199AA8.x, -D_DF4260_80199AA8.y, 0.0f, 1);
+    Matrix_SetGfxMtx(&gMasterDisp);
+    Animation_GetFrameData(D_602D400, 0, boss->vwork);
+    Animation_DrawSkeleton(1, D_602D5AC, boss->vwork, func_DF4260_8018EF38, NULL, &boss->index, &gIdentityMatrix);
+}
+
+extern UNK_TYPE D_603E924;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:2449]: this has type f32[].

File: src/overlays/ovl_DF4260/fox_co.c [hunk 14]
```diff
@@ -0,0 +1,3333 @@
...
+    return 0;
+}
+
+extern AnimationHeader D_602AA7C[];
+extern Limb* D_602AB48[];
+
+void func_DF4260_8018F31C(Object_80* obj80) {
+    Vec3f sp28[10];
+
+    Animation_GetFrameData(D_602AA7C, 0, sp28);
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:2506]:
```suggestion
extern AnimationHeader D_602AA7C;
extern Limb* D_602AB48[];

void func_DF4260_8018F31C(Object_80* obj80) {
    Vec3f sp28[10];

    Animation_GetFrameData(&D_602AA7C, 0, sp28);
```
An AnimationHeader is a single struct.

File: src/overlays/ovl_DF4260/fox_co.c [hunk 15]
```diff
@@ -0,0 +1,3333 @@
...
+    f32 sp54;
+    f32 var_fv1;
+    f32 sp4C;
+    f32 temp_fa0;
+    f32 temp_fa1;
+
+    player->wings.unk_10 = 0.0f;
+    player->wings.unk_08 = 0.0f;
+    player->wings.unk_0C = 0.0f;
+    player->wings.unk_04 = 0.0f;
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:3088]: chain assign?

File: src/overlays/ovl_DF4260/fox_co.c [hunk 16]
```diff
@@ -0,0 +1,3333 @@
...
+            Math_SmoothStepToF(&player->camAt.y, D_DF4260_8019B6D8[63], 0.05f, 500.0f, 0.0f);
+            Math_SmoothStepToF(&player->camAt.z, D_DF4260_8019B6D8[64], 0.05f, 500.0f, 0.0f);
+            temp_fa0 = player->pos.x - D_DF4260_8019B6D8[62];
+            temp_fa1 = player->pos.z - D_DF4260_8019B6D8[64];
+            if (D_80177A80 < 30) {
+                var_fv1 = Math_SmoothStepToAngle(&player->unk_0E8, Math_RadToDeg(-Math_Atan2F(temp_fa0, temp_fa1)),
+                                                 0.5f, 4.0f, 0.0001f) *
+                          20.0f;
+            } else {
+                var_fv1 = Math_SmoothStepToAngle(&player->unk_0E8, Math_RadToDeg(Math_Atan2F(temp_fa0, temp_fa1)), 0.5f,
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:3145]: could a temp be used here for ` Math_RadToDeg(Math_Atan2F(temp_fa0, temp_fa1))` to clean this up a bit?

File: src/overlays/ovl_DF4260/fox_co.c [hunk 17]
```diff
@@ -0,0 +1,3333 @@
...
+    player->pos.x += player->vel.x;
+    player->pos.y += player->vel.y;
+    player->pos.z += player->vel.z;
+    player->unk_138 = player->pos.z;
+    player->unk_0F8 = player->unk_0EC + player->unk_12C + player->unk_130;
+    player->unk_088 += 10.0f;
+    player->unk_080 = -__sinf(player->unk_088 * M_DTOR) * 0.3f;
+    player->unk_0F4 += 8.0f;
+    player->unk_0F0 = __sinf(player->unk_0F4 * M_DTOR);
+}
```
Comment (petrie911) [inline src/overlays/ovl_DF4260/fox_co.c:3333]:
```suggestion
}

```

File: yamls/us/overlays.yaml [hunk 18]
```diff
@@ -8,17 +8,21 @@
     symbol_name_format: DF4260_$VRAM
     subsegments:
     - [0xDF4260, c, DF4260]
```
Comment (petrie911) [inline yamls/us/overlays.yaml:10]: this should be named fox_i1 I think.

**Review by petrie911 — COMMENTED** [review]
Nice work. A few mostly style-related things. Some of them also apply to other places in the file.

File: yamls/us/overlays.yaml (hunk 18 above)
Comment (sonicdcer) [reply yamls/us/overlays.yaml:10]: only if the file split has been done

Comment (sonicdcer) [reply yamls/us/overlays.yaml:10]: which doesn't seem to be the case, i'll do it after merge

File: src/overlays/ovl_DF4260/fox_co.c [hunk 19]
```diff
@@ -0,0 +1,3333 @@
...
+
+void func_DF4260_801887AC(Boss* boss) {
+    f32 sp44;
+    f32 sp40;
+    f32 sp3C;
+    s32 sp38;
+    s32 temp_v0;
+
+    if ((D_801613A0 >= 500)) {
+        temp_v0 = boss->swork[30];
```
[the remark this answers is not in the document]
Comment (inspectredc) [reply src/overlays/ovl_DF4260/fox_co.c:365]: sadly it gives regalloc and couldnt figure out how to get it to match without

File: src/overlays/ovl_DF4260/fox_co.c (hunk 8 above)
Comment (inspectredc) [reply src/overlays/ovl_DF4260/fox_co.c:1367]: it looks like it, maybe thers another way to match though

File: src/overlays/ovl_DF4260/fox_co.c (hunk 14 above)
Comment (inspectredc) [reply src/overlays/ovl_DF4260/fox_co.c:2506]: oh ok, thought someone mentioned it being an array

**Pull request merged**